models/ContainerPage.php

```diff
@@ -67,10 +69,65 @@ public function rules()
     {
         $rules = $this->defaultRules();
         $rules[] = ['in_new_window', 'integer'];
-        $rules[] = [['page_content'], 'safe'];
+	    $rules[] = [['title'], 'filter', 'filter' => array($this, 'stripTagsFilter')];
+	    $rules[] = [['page_content'], 'safe', 'when' => array($this, 'isNotHtml')];
+	    $rules[] = [['page_content'], 'filter', 'when' => array($this, 'isHtml'), 'filter' => array($this, 'purifyFilter')];
+
         return $rules;
     }
 
+	/**
+	 * @param ContainerPage $model the HTML code container
+	 * @return bool whether it is an html type container
+	 */
+	public function isHtml($model)
+	{
+		return $model->type == Container::TYPE_HTML;
+	}
+
+	/**
+	 * @param ContainerPage $model the HTML code
+	 * @return bool whether it is not an html type container
+	 */
+	public function isNotHtml($model)
+	{
+		return $model->type != Container::TYPE_HTML;
+	}
+
+	/**
+	 * strip the title from html tags
+	 * if all is stripped, the title will be "Unnamed_" + a random string, might find a better naming
+	 *
+	 * @param string $title the Title to be stripped
+	 * @return string the stripped title
+	 * @throws \yii\base\Exception
+	 */
+	public function stripTagsFilter($title)
```

Comment:
I'd rename this method. The name indicates it only strips tags, but it actually strips tags and potentially returns a random title name. 
indicating that in the name should make it easier for other developers that read the code to see what happens (right now one might thing it only does stripping the tags) 

while naming is hard: maybe something like `computeTitleFilter`? - still not good, but it indicates that it might do more. 

on the other hand I am wondering if we even should allow empty titles? The title is used on the menu sidebar, right?

Reply:
Yeah, actually the title is a required field when submitting a new/edit page, it does not allow required field to be empty. 
The title gets rendered in the the sidebar as well as in the title tag as "title - Coinsence"

Reply:
About the naming, figured it wouldn't be suitable, still went ahead with it but I'll change it this morning.

Reply:
I think `reformatFilter` is good.

Reply:
:+1: